qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -105,58 +246,97 @@ def is_empty(self) -> bool:
         """Returns whether this tensor is empty or not."""
         return len(self) == 0
 
-    def __getitem__(self, __k: str) -> (np.ndarray | Number):
-        """
-        Returns value matrix in the `PolynomialTensor`.
+    @_optionals.HAS_SPARSE.require_in_call
+    def contains_sparse(self) -> bool:
+        """Returns whether this tensor contains any sparse matrix."""
+        return any(isinstance(self[key], SparseArray) for key in self)
+
+    def __getitem__(self, __k: str) -> (np.ndarray | SparseArray | Number):
+        """Gets the value from the ``PolynomialTensor``.
 
         Args:
             __k: operator key string in the `PolynomialTensor`.
+
         Returns:
-            Value matrix corresponding to the operator key `__k`
+            Value corresponding to the operator key `__k`.
         """
         return self._data.__getitem__(__k)
 
     def __len__(self) -> int:
-        """
-        Returns length of `PolynomialTensor`.
-        """
+        """Returns the length of the ``PolynomialTensor``."""
         return self._data.__len__()
 
     def __iter__(self) -> Iterator[str]:
-        """
-        Returns iterator of the `PolynomialTensor`.
-        """
+        """Returns an iterator of the ``PolynomialTensor``."""
         return self._data.__iter__()
 
+    @_optionals.HAS_SPARSE.require_in_call
+    def todense(self) -> PolynomialTensor:
+        """Converts all internal matrices to dense numpy arrays."""
+        dense_dict: dict[str, ARRAY_TYPE] = {}
+        for key, value in self._data.items():
+            if isinstance(value, SparseArray):
+                dense_dict[key] = value.todense()
+            else:
+                dense_dict[key] = value
+        return PolynomialTensor(dense_dict, validate=False)
+
+    # TODO: change the following type-hint if/when SparseArray dictates the existence of from_numpy
+    @_optionals.HAS_SPARSE.require_in_call
+    def tosparse(
+        self, *, sparse_type: Type[COO] | Type[DOK] | Type[GCXS] = COO
```

Comment:
Maybe this can be made clear in the doctring. This only changes any dense matrices to the given sparse type. Any sparse matrices already existing are unaffected - so you cannot use this for sparse conversions among those types.